### ISSUE TITLE
1.8.2.0 Pre-release

### ISSUE THREAD

**Issue by DillonN** [nicehash/NiceHashMiner#487]
[1.8.2.0 pre-release](https://github.com/nicehash/NiceHashMinerLegacy/releases/tag/1.8.2.0-Pre) is out! The major change is automatic dcri tuning for ClaymoreDual algorithms

I've been working on this on and off for quite some time so I'm glad to get it out. There are some basic instructions on how to get it going in the release page but I'll be posting a Wiki page soon with more in-depth instructions. Basically, you can enable dcri tuning for each dual algorithm by right clicking on it in the benchmark screen and hitting the "Tuning Enabled" option. 

Benchmarking does take a very long time right now, especially if you have many cards. If you have multiple identical cards I recommend tuning on one then copying the settings to the others by right clicking on them in the device list and using that option. As noted in the release page though, your address is used for the benchmarking so you will be paid for it

1.8.2.0 will contain some other features as well. Notably I want to get threaded benchmarking up and running to lower times for users with multiple GPUs (i.e. different GPUs run benchmarks simultaneously). Hopefully that will be in the next pre-release but can't make promises on it.

As usual please restrict discussion here to 1.8.2.0 relevant talk, if you have a suggestion/bug/question that is not related to 1.8.2.0 features specifically please open a new issue. 

To contributors looking to make 1.8.2.0 specific pull requests: it is currently contained on the cd_tuner branch so you can make your PRs for that

**Comment by sabaruddin** [comment]
Great job dude!.. is it also autotune for top profit during mining ?.. or only when benchmark?

**Comment by kansv1** [comment]
Starting with the 1.8.1.4 release and in this version 1.8.2.0-pre in the main window no displays the statistics xmr-stack-cpu. When you start the miner appears for one second and immediately resets to zero. In 1.8.1.4-pre3 everything is fine. Please resolve this problem.

**Comment by adem4ik** [comment]
@sabaruddin I've not checked by myself yet, but according to the tooltip it does autotune during mining:

> "If enabled, NHML will benchmark through all listed dcri values and store the speeds.\nNHML will then use the most profitable speed combination for mining.\nThis mode ignores the -dcri Extra Launch Paramater.",

**Comment by streaml1ne555** [comment]
> Starting with the 1.8.1.4 release and in this version 1.8.2.0-pre in the main window no displays the statistics xmr-stack-cpu. When you start the miner appears for one second and immediately resets to zero. In 1.8.1.4-pre3 everything is fine. Please resolve this problem.

Same issue here. xmrig works fine however. Also for xmrig you may want to publicize use of the --cpu-affinity option. It makes a big difference in performance.

**Comment by Lifeless-Ditto** [comment]
Is there anyway to keep previous benchmark statistics? Or do i have to benchmark it every time the Nicehash Miner Legacy is updated?

**Comment by luongk** [comment]
@Noobperson https://github.com/NiceHash/NiceHashMinerLegacy/wiki/Getting-started#upgrading

**Comment by pducharme** [comment]
Also got the issue of CryptoNight displayed at "0" when using XMR-Stak-CPU.  I switched to XMRig.  Last time I tried, I was not getting good hasrate, but now, it seems to get me about the same hashrate.

**Comment by DillonN** [comment]
@sabaruddin yes the most profitable dcri value will be used while mining

Thanks for reporting the xmr-stak-cpu issue, looks like a problem in 1.8.1.4 but I think I can see the issue and shouldn't be a hard fix. Probably have it in 1.8.1.5 and 1.8.2.0-pre2

**Comment by Kayfolom** [comment]
Copying the settings of the GPU does not work? I tried copying - nothing happens, it is necessary as before to copy a section in a file ""AlgorithmSettings": ["

**Comment by medicnick83** [comment]
Hi all, I'm not sure if this has been mentioned before, but I changed to version "1.8.2.0-Pre" yesterday on most of my rigs (about 4 of the 5 of them) and now I see that CPU mining is almost always at 0, even after I've done a benchmark. I'm thinking maybe it's just a graphical error, but when I close / stop the app, it's as if it was doing nothing, but previously, it's like the PC would have to grind to a halt (if you get what I'm saying)

**Comment by Josephur** [comment]
Recently added a second GTX 1070 to my rig, upgraded to NHML-1.8.2.0 but my 7700k CPU sits there and benchmarks forever (even on fast bench mode).  I'll stick to 1.8.1.4-Pre3 for now, I know you can "skip" benching but it just seems bugged.

**Comment by drunkovsky** [comment]
you can ask on the form to tick "show each graphics card separately" so that you can see the status of each video card when they work on the same algorithm

**Comment by FrankMycHo** [comment]
Cryptonight on ccminer (Nvidia) doesn't seems to work this morning ... anyone else got constant restarting ?
I have low to no hashrate and the miner restart every minute or so

**Comment by kansv1** [comment]
Yes. The miner ccminer (Nvidia) restart every minute.

**Comment by drunkovsky** [comment]
Today I have and on the beta version and on older versions everything fell away

**Comment by drunkovsky** [comment]
now works more stably on all versions

**Comment by RileyMShea** [comment]
Would be nice if the GPU's could be auto-dectected or selected from a dropdown and the DCRI values from the wiki [https://github.com/nicehash/NiceHashMinerLegacy/wiki/Tweaking-ClaymoreDual-Intensity](url) were used as a baseline for benchmarking. Testing something like +/- 25% from the wiki's DCRI values seem like a good starting point?

**Comment by DillonN** [comment]
The cryptonight issues were a server problem

The 0H/s for xmr-stak-cpu was fixed in 1.8.1.5 and will be in the pre2

@RileyMShea the values on that page are for older versions of ClaymoreDual and not particularly accurate anymore. With so many cards available it would be difficult to set default values that worked for everyone

**Comment by browned** [comment]
Not sure this is an issue but I have been having trouble with a small setup with a modded RX470 and a standard RX570. I have been tuning them for a while with previous NHML editions and found the RX470 liked to use asm -2, dcri -16 to 20. The RX570 likes to use asm 2, dcri 50. These settings consistently pull more BTC a day than any other setting with daggerlbry or dagger/dcrd.

Using 1.8.2.0, precise benchmark dcri from 10 to 70 (steps of 2) the RX470 runs with dcri at 28 to 30, and the RX570 runs at dcri 60 to 62. These settings result in .50c less per day per card, yes BTC is less as well.

Running both the cards at once results in the RX470 mining eth a lot lower and lbry a lot high, almost like the dcri values are not applying correctly and it is using the RX570 settings.

Can anyone confirm this is happening them as well?

Does NHML benchmark with asm -2?
Does NHML benchmark at the same DAG size, I have noticed recently the DAG size changing to 4, 160, 150, all within minutes?

**Comment by MahmoudiOussama** [comment]
Can we edit the donation percentage for xmr miner ?
Also nicehash switched a lot between different algorithms, it has to be more stable.

**Comment by davidschou** [comment]
Not sure if it's my system or Nice Hash Legacy 1.8.2.0-Pre.  I have a GTX 980 and Vega 56 (w/ 64 bios).  I've noticed that the monitor will flicker off then back on.  Wattman settings  gets the "unexpected system failure" and loses it's settings.  I also noticed that the Mining Rate goes to 0 and finds a new algo.  

Not sure if it's GPU triggering it, or NHML triggering it.  Just wanted to pass this one, let me know if you need logs and what to provide to help. 

AMD Driver: 17.11.2 Crimson Relive (not the blockchain one - that one also freq lost wattman settings) (i've uninstalled MSI and don't have any other OC tool)

Edit: Adding comment: NHML will switch Algorithms without the monitor flashing/resetting Wattman settings - so it's not 100% of the time I am seeing this.

**Comment by streaml1ne555** [comment]
Are you overclocking the Vega? If so it might be too aggressive an overclock. The flickering monitor and algorithm switching is typical of a driver crash which is usually OC related.

**Comment by bonifacio123** [comment]
UPDATE / FIXED: Seems my issues were due to Windows Remote Desktop. Disabling printer support fixed the issue. 

1.8.2.0-pre2 is causing my test setup to shut down or blue screen whereas pre1 runs just fine. Seeing lots of this in the log

`[2017-11-27 10:12:18] [INFO] [CPUDIAG] System.InvalidOperationException: Cannot load Counter Name data because an invalid index '' was read from the registry.
   at System.Diagnostics.PerformanceCounterLib.GetStringTable(Boolean isHelp)
   at System.Diagnostics.PerformanceCounterLib.get_NameTable()
   at System.Diagnostics.PerformanceCounterLib.get_CategoryTable()
   at System.Diagnostics.PerformanceCounterLib.CounterExists(String category, String counter, Boolean& categoryExists)
   at System.Diagnostics.PerformanceCounterLib.CounterExists(String machine, String category, String counter)
   at System.Diagnostics.PerformanceCounter.InitializeImpl()
   at System.Diagnostics.PerformanceCounter.Initialize()
   at System.Diagnostics.PerformanceCounter.NextSample()
   at System.Diagnostics.PerformanceCounter.NextValue()`

Here's the log file:

[log.txt](https://github.com/nicehash/NiceHashMinerLegacy/files/1506714/log.txt)

**Comment by adem4ik** [comment]
Can't finish benchmark for XMR-stak on my CPU. It seems like xmr-stak.exe's console window doesn't even want to start.

**Specs:** Windows 10 x64, Intel Core i5-2320 Sandy Bridge, 8 GB RAM, 4 GB SWAP, NVIDIA GeForce GTX 1060 6GB

**Prominent messages from logs:**
**/logs/log.txt: ** [logs.zip](https://github.com/nicehash/NiceHashMinerLegacy/files/1508068/logs.zip)
```
[2017-11-28 03:53:16] [INFO] [CPUDIAG] System.InvalidOperationException: Category doesn't exist.
   в System.Diagnostics.PerformanceCounterLib.CounterExists(String machine, String category, String counter)
   в System.Diagnostics.PerformanceCounter.InitializeImpl()
   в System.Diagnostics.PerformanceCounter.Initialize()
   в System.Diagnostics.PerformanceCounter.NextSample()
   в System.Diagnostics.PerformanceCounter.NextValue()
   в NiceHashMiner.Devices.CPUComputeDevice.get_Load() в C:\Users\Dillon\Documents\Code\NiceHashMiner\NiceHashMiner\Devices\ComputeDevice\CPUComputeDevice.cs:string 18
```
**xmr-stak/0-0_log.txt:** [0-0_log.txt](https://github.com/nicehash/NiceHashMinerLegacy/files/1508080/0-0_log.txt)
`[2017-11-28 03:51:35] : SOCKET ERROR - [] CONNECT error: Pool port number not specified, please use format <hostname>:<port>.`

**Comment by E14** [comment]
BitDefender finds a "trojan.generickd.12632180" in the archive from https://github.com/nicehash/NiceHashMinerLegacy/releases/download/1.8.2.0-Pre2/NHML-1.8.2.0-Pre2.zip

Notably, it does _not_ find it in the 1.8.1.... stable release

**Comment by DillonN** [comment]
@adem4ik interesting, it appears xmr-stak isn't getting the port number from the config file. Can you check the config file bench_0-0.txt and see that it shows the proper pool address config? E.g. `"cryptonight.eu.nicehash.com:3355"`. And it's only a problem during benchmarking? 

The error CPUDIAG error is just from NHML trying to poll CPU usage, shouldn't cause an issue with anything except seeing CPU usage on the stats webpage

@E14 see [wiki](https://github.com/NiceHash/NiceHashMinerLegacy/wiki/Troubleshooting#nicehash-miner-legacy-or-included-miners-are-being-flagged-by-anti-virus)

**Comment by adem4ik** [comment]
@DillonN yes, the `"pool_address"` is empty in the **bench_0-0.txt**, while there is a proper `"pool_address": "cryptonight.eu.nicehash.com:3355"` in the **config.txt**. And yes, it is the only noticeable problem during benchmarking.

**Comment by SecurityImportant** [comment]
Whilst running this release a trojan was injected into the NiceHashLegacy.exe after a day of running. Trojan:win32/Tilken.B!cl

Interesting? Seems there are a lot of reports online recently regarding such related to mining software!

**Comment by CreamyG31337** [comment]
definitely some bugs with the benchmarking - sometimes the benchmark window stays open 'benchmarking' even when nothing is going on. sometimes I hit benchmark after clearing a speed and it refuses to benchmark, insisting that it's 'complete' even with the row empty.
i think this is new since the multi-threaded benchmarking was added.

**Comment by Balzhur** [comment]
There is '=' missing in the MinerOptionPackage_Xmrig.json in --cpu-affinity and --cpu-priority option definitions. Please fix.

**Comment by techman05** [comment]
Hi I have an issue. with pre 2. I  was going through doing the individual   benchmarks and After I had a few I stopped and tried to start without finishing the rest and at the moment its only running the cpu and not my amd card. Should I have to do all benchmarks for it to start running?

**Comment by rpm604** [comment]
Hi DillonN, posting here as recommended! Sorry for the error. From my original post:

"I'm getting two very specific rejected shares right now.

Rig 1:

• 1724 / 1773 (97.2%)
• 41 job not found
• 8 invalid nonce

Rig 2:

• 1880 / 1935 (97.2%)
• 41 job not found
• 14 invalid nonce

As in the title, 5 Vegas across 2 rigs, using xmr-stak, in the actual NiceHash app."

This is the content of my config_nh:

// This config file was autogenerated by NHML.
// The values were filled from config.txt. If you wish to edit them, you should edit that file instead.

  "pool_list": [
    {
      "pool_address": "cryptonight.usa.nicehash.com:3355",
      "wallet_address": "3HPSXTGRJRz5KVEavPjS9aawS2uTC8HZ5i.worker1",
      "pool_password": "x",
      "use_nicehash": true,
      "use_tls": false,
      "tls_fingerprint": "",
      "pool_weight": 1
    }
  ],
  "currency": "monero",
  "call_timeout": 10,
  "retry_time": 30,
  "giveup_limit": 0,
  "verbose_level": 4,
  "print_motd": true,
  "h_print_time": 60,
  "aes_override": null,
  "use_slow_memory": "warn",
  "tls_secure_algo": true,
  "daemon_mode": false,
  "flush_stdout": false,
  "output_file": "",
  "httpd_port": 4000,
  "http_login": "",
  "http_pass": "",
  "prefer_ipv4": true

**Comment by riccardoruspoli** [comment]
Xmr-Stak gives a wrong value after the benchmark, like ~5232 kH/s. The correct value is ~523.2 H/s, so the correct result of the benchmark is one tenth of the result. This happens on 3 different rigs with different GPUs.

**Comment by rpm604** [comment]
Issue above remains. Any word on it? These rejected shares only happen on the nicehash pool, when direct pool mining to other pools it is not an issue.

**Comment by KylanRace** [comment]
I'm getting the exact same problem as @rpm604 . Though I seem  to get anywhere from 16%-25% rejected shares lately, regardless of  the server  location that I choose.

**Comment by AndrewJP23** [comment]
Hi, im having 2 problems, first is the benchmarks are not completing (+ Decred, +Lbry, + Pascal, + Sia, CryptoNight (XmrStak), CryptoNight (CLaymore), Equihash (Optiminer), Sia (Prospector), and Skunk (Prospector)), i have run all of them by themselves and as a group and it shows as being completed but after running through them the 9 still show as unbenchmarked. Also, for one of the benchmarks i get a " AMD HSA Code Object Failed Loading" error code. I have used both 1.8.1.5 and 1.8.2.0 with the same 9 not working. Second issue is that when i try to mine none of them actually initialize. The error either says "job timeout, retry in 20 secs" or (with sgminer) it shows the first 2 lines of text for a few minutes, closes, and reopens automatically. I have cleared the DNS cache (when i tried googling for help), switching to different servers, and again, using 2 different versions, and amd drivers are up to date. Windows 10 1x Vega 64

**Comment by techman05** [comment]
Andrew do the  benchmark on the non preview version and copy/paste the configs folder. Thats what I did to get past issues doing the benchmark.  See if rolling back to an older driver if you just started having issues.

**Comment by AndrewJP23** [comment]
non preview version? And i havent been able to mine since nicehash came back, and to my knowledge no new driver dropped for it

**Comment by techman05** [comment]
The release version. Every driver except for the last 2 seemed to work. That's my current thought

I ran my benchmarks on the release version and the only issue I've had so far is installing the current firmware version.  Either way I can hope there should be an update soon to work with the presumed updated nice hash page.

**Comment by AndrewJP23** [comment]
the benchmarks fifnt work in 1.8.1.5, so i went to 1.8.1.4 and ran them, and then copied and pasted the configs and that fixed all of my problems (miner is now mining), thank you very much!

**Comment by borimol** [comment]
This Legacy version is abandonware? No updates since 27 nov.

**Comment by techman05** [comment]
They probably have them working on other projects. Probably alt coins and web page management are higher priority at the moment.

**Comment by mifurman** [comment]
I have been trying (after reading many posts) to setup Stak-XMR on my Vega 64 and it is still crashing during benchmark. When running Claymore XMR, I am getting 1700 Hs CryptoNight. Does anyone have a example of amd file (xmr-stak folder), please?

**Comment by mgrycow** [comment]
Hi. I'm a bit new to this so maybe I don't get it, but why can we only run cryptonight on the cpu? I mean, the algorithm on the GPU regularly switches away from cryptonight, wouldn't this mean at those times it would be more profitable to switch the cpu as well? I would greatly appreciate it if you'd put more in. Am I alone in this? 

Also, can anyone tell me I his pre release is worth getting instead of 1.8.1.6? 

Thx for any help understanding all of this.

**Comment by drunkovsky** [comment]
There is a request. You can make the parameter "delayed launch" in new versions of the program, so you can specify the time in seconds, how long the program should wait before starting its work.

What is it for.
Sometimes on the rigs where there are many video cards, the program runs earlier than windows has time to start all the video cards, and the program says that there are no video cards.
We have to run the NiceHashMinerLegacy program in cmd, indicating the delayed start time.

**Comment by Braintelligence** [comment]
@mgrycow The implications of if you should use 1.8.2.0 pre2 or not are written in the release note. If your GPUs switch from cryptonight to something else, then this doesn't have to mean that your CPU should also switch. The reason for this is the same reason why different GPUs work on or switch to different algorithms.

**Comment by adem4ik** [comment]
@DillonN I have some questions - are you going to continue developing 1.8.2.0? will **cd_tuner** branch be merged with the **master** branch some day? I'm curious cause I have made some pull requests with ru-translation previously to **cd_tuner** branch, so those translations are missing in the **master** branch.

**Comment by S74nk0** [comment]
Issue related to old nhm version/build. Please use latest version and open a new issue for any bugs or feature requests.
